docs/reference/function-configuration/code-entry-types.md

```diff
@@ -0,0 +1,245 @@
+- `archive` &mdash; download the code as a ZIP archive file from an Iguazio Data Science Platform data container or from any URL that doesn't require download authentication. See [Archive-file code-entry type (`archive`)](#code-entry-type-archive).
+- `s3` &mdash; download the code as a ZIP archive file from an AWS S3 bucket. See [AWS S3 code-entry type (`s3`)](#code-entry-type-s3).
+
+Additional information for performing the download &mdash; such as the download URL or authentication information &mdash; is provided in dedicated configuration fields for each code-entry type, as detailed in the documentation of each code-entry type.
+
+> **Note:**
+> - When `spec.image` or `spec.build.functionSourceCode` are set, `spec.build.codeEntryType` is ignored. See [Determining the code-entry type](#code-entry-type-determine).
+> - The downloaded code files are saved and can be used by the function handler.
+
+> **Dashboard Note:** To configure an external function-code source from the dashboard, select the relevant code-entry type &mdash; `Archive`, `GitHub`, or `S3` &mdash; from the **Code entry type** list.
+
+The downloaded function code can optionally contain a **function.yaml** file with function configuration for enriching the original configuration (in the configuration file that sets the code-entry type) according to the following merge strategy:
+
+- Field values that are set only in the downloaded configuration are added to the original configuration.
+- List and map field values &mdash; such as `meta.labels` and `spec.env` &mdash;are merged by adding any values that are set only in the downloaded configuration to the values that are set in the original configuration.
+- In case of a conflict &mdash; i.e., if the original and downloaded configurations set different values for the same element &mdash; the original configuration takes precedence and the value in the downloaded configuration is ignored.
+
+<a id="code-entry-type-github"></a>
+### GitHub code-entry type (`github`)
+
+Set the [`spec.build.codeEntryType`](/docs/reference/function-configuration.md#spec.build.codeEntryType) function-configuration field to `github` (dashboard: **Code entry type** = `GitHub`) to download the function code from a GitHub repository. The following configuration fields provide additional information for performing the download:
+
+- `spec.build` &mdash;
+  - `path` (dashboard: **URL**) (Required) &mdash; the URL of the GitHub repository that contains the function code.
+  - `codeEntryAttributes` &mdash;
+    -  `branch` (dashboard: **Branch**) (Required) &mdash; the GitHub repository branch from which to download the function code.
+    - `headers.Authorization` (dashboard: **Token**) (Optional) &mdash; a GitHub access token for download authentication.
+    - `workDir` (dashboard: **Work directory**) (Optional) &mdash; the relative path to the function-code directory within the configured repository branch.
```

Comment:
If optional then I recommend stating the default behaviour - IIRC if this is not given we default to the root of the target git repository

Reply:
I initially debated whether to add the default root-directory info and decided it wasn't necessary, but given your comment I've now edited all `workDir` descriptions (for `github`, `archive`, and `s3`) to explicitly refer to the default + I slightly edited the base description for the archive-file types. See commit b7b08e4.